typewise-alert.c

```diff
@@ -11,61 +12,74 @@ BreachType inferBreach(double value, double lowerLimit, double upperLimit) {
   return NORMAL;
 }
 
-BreachType classifyTemperatureBreach(
-    CoolingType coolingType, double temperatureInC) {
-  int lowerLimit = 0;
-  int upperLimit = 0;
-  switch(coolingType) {
-    case PASSIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 35;
-      break;
-    case HI_ACTIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 45;
-      break;
-    case MED_ACTIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 40;
-      break;
-  }
-  return inferBreach(temperatureInC, lowerLimit, upperLimit);
+BreachType classifyTemperatureBreachforPassiveCooling(CoolingType coolingType, double temperatureInC) 
+{
+  if(coolingType==PASSIVE_COOLING) 
```

Comment:
the lower and upper range values for the cooling type varies. using this, maybe you could reduce the number of checks by maintaining a list of pair of lower/upper values , one each for different cooling types. then use the list to get the lower/upper range for given cooling type.

Reply:
I have tried to re-modify the code as you mentioned in the comment. Kindly let me know if I have to rework on any other points. Your feedback will be really helpful for me to correct my approach. Thank you.